docs/nomad/README.md

```diff
@@ -0,0 +1,250 @@
+# HashiCorp Nomad Hetzner Cloud csi-driver
+
+## Preconditions
+
+- Nomad >= 1.4.x cluster installed and running (tested on Nomad version 1.5.x).
+- The HCL resources are meant to be executed on a machine having nomad installed (with access to the Nomad API).
+
+## Getting Started
+
+### CSI Setup
+
+1. Create a read+write API token in the [Hetzner Cloud Console](https://console.hetzner.cloud/).
+
+2. Create a Nomad Variable for the HCLOUD token:
+
+> [!NOTE]
+> Consider using HashiCorp Vault for secrets management, see https://developer.hashicorp.com/nomad/docs/job-specification/template#vault-kv-api-v2
+
+```sh
+export HCLOUD_TOKEN="..."
+nomad var put secrets/hcloud hcloud_token=$HCLOUD_TOKEN
+```
+
+3. Create a CSI Controller Job
+
+```hcl
+# file: hcloud-csi-controller.hcl
+
+job "hcloud-csi-controller" {
+  datacenters = ["dc1"]
+  namespace   = "default"
+  type        = "service"
+
+  group "controller" {
+    count = 2
```

Comment:
Why are 2 controllers needed? 🤔

Reply:
The thinking behind this is, one thing we want to make sure that controller is always available. If the node goes down the controller is running on, a volume won't be able to be migrated, right?

That is why we defined 2 of the controllers to run and also defined:

```
    constraint {
      distinct_hosts = true
    }
```

To ensure, controller instances run on different nodes.

Reply:
If having 2 controllers is a good idea (I'll let others judge if this is valid) in this documentation, maybe you should add a comment above the field with your explanation ? If one person already raised a question about this, maybe others will.

Reply:
The csi-controller is a stateless gRPC service. There should be no issue running two of them, as long as the caller makes sure to not send duplicate requests (and even those should be handled properly). I do not know if Nomad utilizes multiple controllers.


---

If one deploys this to a single node Nomad cluster, will the Job show an error because only a single task could be created?

Reply:
> If having 2 controllers is a good idea (I'll let others judge if this is valid) in this documentation, maybe you should add a comment above the field with your explanation ? If one person already raised a question about this, maybe others will.

makes sense. 

We run this in prod and works as expected. I remember a "pressure" on our cluster when running only one instance of controller in "node down running csi controller" situation. 

> If one deploys this to a single node Nomad cluster, will the Job show an error because only a single task could be created?

I am not sure about this, I would say that it would work with a warning. However, single node clusters are considered experimental for testing. We refer to HashiCorp's "Best Practice for Production" with 3 master nodes. We should point that out in the doc.